document/document.go

```diff
@@ -86,16 +86,108 @@ func (fb FieldBuffer) GetByField(field string) (Value, error) {
 	return Value{}, ErrFieldNotFound
 }
 
+// SetFieldValue replaces a field if it already exists or creates one if not.
+func (fb *FieldBuffer) SetFieldValue(field string, reqValue Value) error {
```

Comment:
I don't think it's worth it to export this method. Also, this only basically does an if and an else, I don't think it deserves being a function.

Reply:
Ooh ! The parser is a regression, I don't know what I did bad. So I going to correct it.